src/wallet/crypter.cpp

```diff
@@ -15,21 +15,40 @@
 #include <openssl/aes.h>
 #include <openssl/evp.h>
 
-bool CCrypter::SetKeyFromPassphrase(const SecureString& strKeyData, const std::vector<unsigned char>& chSalt, const unsigned int nRounds, const unsigned int nDerivationMethod)
+bool CCrypter::SetKeyFromPassphrase(
+        const SecureString& strKeyData,
+        const std::vector<unsigned char>& chSalt,
+        const unsigned int nRounds,
+        const unsigned int nDerivationMethod,
+        const std::vector<unsigned char> vchOtherDerivationParameters)
 {
     if (nRounds < 1 || chSalt.size() != WALLET_CRYPTO_SALT_SIZE)
         return false;
 
-    int i = 0;
-    if (nDerivationMethod == 0)
-        i = EVP_BytesToKey(EVP_aes_256_cbc(), EVP_sha512(), &chSalt[0],
-                          (unsigned char *)&strKeyData[0], strKeyData.size(), nRounds, chKey, chIV);
+    if (nDerivationMethod == 0) {
```

Comment:
And also we're removing the original derivation method completely rather than providing a migration pathway. So I think it's sufficient to say that someone who used `-developerencryptwallet` should export their wallet on the old version and then re-import it into an empty new wallet after upgrading. I'm not even sure we should say that in the release notes, because technically with this PR merged we'd still have encryption disabled via the flag, and writing in the release notes would just draw attention to it while it is still completely unsupported.